src/models/transfer/facade.js

```diff
@@ -1088,6 +1090,21 @@ const transferStateAndPositionUpdate = async function (param1, enums, trx = null
   }
 }
 
+const updatePrepareReservedForwarded = async function (transferId) {
+  try {
+    const knex = await Db.getKnex()
+    return await knex('transferStateChange')
+      .insert({
+        transferId,
+        transferStateId: TransferInternalState.RESERVED_FORWARDED,
+        reason: null,
+        createdDate: Time.getUTCString(new Date())
+      })
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
```

Comment:
I do not know very well the state update engine, but this looks a bit unexpected approach to me. My comment is not directly related to the feature addressed by this PR, but what gives us a guarantee that another insert for a different state is not happening at the same time resulting in both inserts being successful?